src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -91,53 +98,79 @@ async function pickProject(
+        })
+        throw new ToolkitError('Transform cancelled', { code: 'DidNotConfirmDisclaimer', cancelled: true })
+    } else {
+        telemetry.codeTransform_jobIsStartedFromUserPopupClick.emit({
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+        })
+    }
 
-        transformByQState.setToRunning()
-        transformByQState.setProjectPath(state.project.description!)
-        sessionPlanProgress['uploadCode'] = StepProgress.Pending
-        sessionPlanProgress['buildCode'] = StepProgress.Pending
-        sessionPlanProgress['transformCode'] = StepProgress.Pending
-        sessionPlanProgress['returnCode'] = StepProgress.Pending
-        const startTime = new Date()
+    transformByQState.setToRunning()
+    transformByQState.setProjectPath(state.project.description!)
+    sessionPlanProgress['uploadCode'] = StepProgress.Pending
+    sessionPlanProgress['buildCode'] = StepProgress.Pending
+    sessionPlanProgress['transformCode'] = StepProgress.Pending
+    sessionPlanProgress['returnCode'] = StepProgress.Pending
+    const startTime = new Date()
+
+    telemetry.codeTransform_jobStartedCompleteFromPopupDialog.emit({
+        codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+        codeTransformJavaSourceVersionsAllowed: CodeTransformTelemetry.toJDKMetricValue(
+            transformByQState.getSourceJDKVersion()
+        ) as CodeTransformJavaSourceVersionsAllowed,
+        codeTransformJavaTargetVersionsAllowed: CodeTransformTelemetry.toJDKMetricValue(
+            transformByQState.getTargetJDKVersion()
+        ) as CodeTransformJavaTargetVersionsAllowed,
+        codeTransformConfigurationFilePath: '', // TODO: set or remove this
+    })
 
-        vscode.commands.executeCommand('workbench.view.extension.aws-codewhisperer-transformation-hub')
-        vscode.commands.executeCommand('aws.amazonq.showPlanProgressInHub', startTime.getTime())
-        vscode.commands.executeCommand('setContext', 'gumby.isStopButtonAvailable', true)
-        vscode.commands.executeCommand('setContext', 'gumby.isTransformAvailable', false)
-        vscode.commands.executeCommand('setContext', 'gumby.isPlanAvailable', false)
-        resetReviewInProgress()
+    vscode.commands.executeCommand('workbench.view.extension.aws-codewhisperer-transformation-hub')
+    vscode.commands.executeCommand('aws.amazonq.showPlanProgressInHub', startTime.getTime())
+    vscode.commands.executeCommand('setContext', 'gumby.isStopButtonAvailable', true)
+    vscode.commands.executeCommand('setContext', 'gumby.isTransformAvailable', false)
+    vscode.commands.executeCommand('setContext', 'gumby.isPlanAvailable', false)
+    resetReviewInProgress()
 
-        await vscode.commands.executeCommand('aws.amazonq.refresh')
+    await vscode.commands.executeCommand('aws.amazonq.refresh')
+    await telemetry.amazonq_codeTransformInvoke.run(async span => {
```

Comment:
Should everything added to this function still be wrapped in this `run`? e.g. `validateProjectSelection(state.project)` used to be a part of this, but now it isn't...does that not constitute a failed run metric?

Reply:
No -- in fact, it was considered a bug, as conceptually the prevalidation happens _before_ the code transformation is attempted. We were getting a large number of false failures by including this section in that metric.